starknet_devnet/starknet_wrapper.py

```diff
@@ -619,13 +619,12 @@ async def __get_query_state(self, block_id: BlockId = DEFAULT_BLOCK_ID):
         assert isinstance(block_id, dict)
         if block_id.get("block_hash"):
             # takes care of parsing
-            block = await self.blocks.get_by_hash(block_id["block_hash"])
-            return self.blocks.get_state(block.block_number)
+            return self.blocks.get_state(int(block_id["block_hash"], base=16))
```

Comment:
The old implementation took care of parsing inside of `get_by_hash` (that's what the comment in L621 says). Now it no longer checks for `block_id["block_hash"]` validity but immediately tries parsing, which might fail.

Reply:
@FabijanC so this should not be:
```
return self.blocks.get_state(int(block_id["block_hash"], base=16))
```
but
```
return self.blocks.get_state(int(block_id.get("block_hash"), base=16))
```
?

Reply:
Fixed in 4e0d79aa7ce421418e021dbd917652b22b7955b3